.github/workflows/ci.yml

```diff
@@ -68,26 +91,34 @@ jobs:
         name: Unshallow
         run: git fetch --prune --unshallow
       -
-        name: Read go version (Unix)
-        if: ${{ runner.os != 'Windows' }}
-        id: go-version-unix
+        name: Read go version
+        id: go-version
+        shell: bash
         run: |
           content=`cat ./.go-version`
           echo "::set-output name=content::$content"
-      -
-        name: Read go version (Windows)
-        if: ${{ runner.os == 'Windows' }}
-        id: go-version-win
-        run: |
-          $content = Get-Content .\.go-version -Raw
-          echo "::set-output name=content::$content"
       -
         name: Set up Go
         uses: actions/setup-go@v2
         with:
           # TODO: Replace with go-version-from-file when it is supported
           # https://github.com/actions/setup-go/pull/62
-          go-version: ${{ steps.go-version-unix.outputs.content || steps.go-version-win.outputs.content }}
+          go-version: ${{ steps.go-version.outputs.content }}
+      - uses: actions/cache@v2
+        with:
+          # In order:
+          # * Module download cache
+          # * Build cache (Linux)
+          # * Build cache (Mac)
+          # * Build cache (Windows)
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            ~/Library/Caches/go-build
+            ~/AppData/Local/go-build
```

Comment:
I wish this was available as some kind of variable from somewhere - it must have been lots of trial & error and/or documentation reading to put all these paths together! 😅 

Am I right that `%LocalAppData%\go-build` ~= `~/AppData/Local/go-build` but we just assume the variable is never going to be set to anything else within the Windows GHA environment? https://github.com/actions/cache/blob/main/examples.md#windows-1

Reply:
I did start out with that example, but https://github.com/mvdan/github-actions-golang#how-do-i-set-up-caching-between-builds lumped them all together to reduce multiple filters. This seems to work in a 'if this folder doesn't exist then don't cache it' sort of way.

The `~/` path thing came from https://github.com/actions/cache/issues/705, which details how little this action is documented for different platforms. I wasn't seeing anything cached for `%LocalAppData%`, but did get for `~/AppData` so I switched. This does kinda make sense, since the default shell on Windows is now pwsh for Github Actions, and pwsh does not use % for environment variables.

We could dig further into this, but https://github.com/actions/cache/issues/301, makes it pretty obvious this action wasn't designed for windows so we should probably abandon using it until that is fixed. It currently takes longer to download and extract the cache than it does to run `go mod download`

Reply:
Makes sense! Thanks for the extra context.